app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt

```diff
@@ -17,46 +17,386 @@
+            mlsKeyPackageCountUseCase().let {
+                when (it) {
+                    is MLSKeyPackageCountResult.Success -> {
+                        state = state.copy(
+                            keyPackagesCount = it.count,
+                            mslClientId = it.clientId.value
+                        )
+                    }
+
+                    is MLSKeyPackageCountResult.Failure.NetworkCallFailure -> {
+                        state = state.copy(mlsErrorMessage = "Network Error!")
+                    }
+
+                    is MLSKeyPackageCountResult.Failure.FetchClientIdFailure -> {
+                        state = state.copy(mlsErrorMessage = "ClientId Fetch Error!")
+                    }
+
+                    is MLSKeyPackageCountResult.Failure.Generic -> {}
+                }
+            }
+        }
+    }
+    //endregion
+}
+//endregion
 
 @Composable
 fun DebugDataOptions(
     appVersion: String,
     buildVariant: String,
-    clientId: String,
+    commitish: String,
+    debugId: String,
     onCopyText: (String) -> Unit
 ) {
+
+    val viewModel: DebugDataOptionsVewModel = hiltViewModel()
+
     Column {
         FolderHeader(stringResource(R.string.label_debug_data))
         SettingsItem(
-            title = appVersion,
+            title = stringResource(R.string.app_version),
+            text = appVersion,
             trailingIcon = R.drawable.ic_copy,
             onIconPressed = Clickable(
                 enabled = true,
                 onClick = { onCopyText(appVersion) }
             )
         )
         SettingsItem(
-            title = buildVariant,
+            title = stringResource(R.string.build_variant_name),
+            text = buildVariant,
             trailingIcon = R.drawable.ic_copy,
             onIconPressed = Clickable(
                 enabled = true,
                 onClick = { onCopyText(buildVariant) }
             )
         )
         SettingsItem(
-            title = clientId,
+            title = stringResource(R.string.label_code_commit_id),
+            text = buildVariant,
             trailingIcon = R.drawable.ic_copy,
             onIconPressed = Clickable(
                 enabled = true,
-                onClick = { onCopyText(clientId) }
+                onClick = { onCopyText(buildVariant) }
             )
         )
+        if (BuildConfig.PRIVATE_BUILD) {
+            SettingsItem(
+                title = stringResource(R.string.debug_id),
+                text = debugId,
+                trailingIcon = R.drawable.ic_copy,
+                onIconPressed = Clickable(
+                    enabled = true,
+                    onClick = { onCopyText(debugId) }
+                )
+            )
+        }
+        if (BuildConfig.PRIVATE_BUILD) {
+            ProteusOptions(
+                isEncryptedStorageEnabled = viewModel.state.isEncryptedProteusStorageEnabled,
+                onEncryptedStorageEnabledChange = { enabled ->
```

Comment:
I would wrap it in `remember` to not recompose when not needed, or change `viewModel.enableEncryptedProteusStorage` to take `enabled` parameter like `viewModel.onEncryptedStorageEnabledChange(enabled)` and then pass it directly: `onEncryptedStorageEnabledChange = viewModel::onEncryptedStorageEnabledChange`
like it's done for other items on this screen